flake.nix

```
@@ -0,0 +1,27 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+  };
+
+  outputs = {nixpkgs, ...}: let
+    system = "x86_64-linux";
+    pkgs = nixpkgs.legacyPackages.${system};
+  in {
+    devShells.${system}.default = pkgs.mkShell {
+      nativeBuildInputs = with pkgs; [
```

suggestion (non-blocking): I'm using

```suggestion
      packages = with pkgs; [
```

instead in my own flake but I failed to understand when to use which based on the documentation. maybe the second paragraph is the relevant distinction?

- https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-mkShell-attributes: `Add executable packages to the nix-shell environment.`
- https://nixos.org/manual/nixpkgs/stable/#var-stdenv-nativeBuildInputs: `But since these packages are only guaranteed to be able to run then, they shouldn’t persist as run-time dependencies. This isn’t currently enforced, but could be in the future.`

---

https://discourse.nixos.org/t/difference-between-buildinputs-and-packages-in-mkshell/60598/2 suggests:

> So, in a nutshell, just use packages and forget the other inputs for mkShell exist.

So let's do that! :P